common/util/Assert.cpp

```diff
@@ -3,21 +3,23 @@
 
 #include "Assert.h"
 #include <string_view>
+#include <common/log/log.h>
 
 void private_assert_failed(const char* expr,
                            const char* file,
                            int line,
                            const char* function,
                            const char* msg) {
   if (!msg || msg[0] == '\0') {
-    fprintf(stderr, "Assertion failed: '%s'\n\tSource: %s:%d\n\tFunction: %s\n", expr, file, line,
-            function);
+    std::string log = fmt::format("Assertion failed: '{}'\n\tSource: {}:{}\n\tFunction: {}\n", expr,
+                                  file, line, function);
+    lg::die(log);
   } else {
-    fprintf(stderr, "Assertion failed: '%s'\n\tMessage: %s\n\tSource: %s:%d\n\tFunction: %s\n",
-            expr, msg, file, line, function);
+    std::string log =
+        fmt::format("Assertion failed: '{}'\n\tMessage: %s\n\tSource: {}:{}\n\tFunction: {}\n",
```

Comment:
`%s` should be `{}` here. (this is part of the reason i don't like libfmt that much... instead of being a compiler error it ends up being some confusing "format error" exception thrown that you only see at runtime)